src/bun.js/bindings/KeyObject.cpp

```diff
@@ -1239,6 +1246,492 @@ JSC::EncodedJSValue KeyObject__createSecretKey(JSC::JSGlobalObject* lexicalGloba
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    auto* key = jsDynamicCast<JSCryptoKey*>(callFrame->argument(0));
+    if (!key) {
+        // No JSCryptoKey instance
+        JSC::throwTypeError(globalObject, scope, "expected CryptoKey as first argument"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    JSValue bufferArg = callFrame->uncheckedArgument(1);
+
+    auto buffer = KeyObject__GetBuffer(bufferArg);
+    if (buffer.hasException()) {
+        JSC::throwTypeError(globalObject, scope, "expected Buffer or array-like object as second argument"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    auto vectorData = buffer.releaseReturnValue();
+    auto& wrapped = key->wrapped();
+    auto key_type = wrapped.type();
+    auto id = wrapped.keyClass();
+
+    auto hash = WebCore::CryptoAlgorithmIdentifier::SHA_256;
+    auto algorithm = callFrame->argument(2);
+    auto customHash = false;
+    if (!algorithm.isUndefinedOrNull() && !algorithm.isEmpty()) {
+        customHash = true;
+        if (!algorithm.isString()) {
+            JSC::throwTypeError(globalObject, scope, "algorithm is expected to be a string"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        auto algorithm_str = algorithm.toWTFString(globalObject);
+        RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+        auto identifier = CryptoAlgorithmRegistry::singleton().identifier(algorithm_str);
+        if (UNLIKELY(!identifier)) {
+            JSC::throwTypeError(globalObject, scope, "digest not allowed"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+
+        switch (*identifier) {
+        case WebCore::CryptoAlgorithmIdentifier::SHA_1:
+        case WebCore::CryptoAlgorithmIdentifier::SHA_224:
+        case WebCore::CryptoAlgorithmIdentifier::SHA_256:
+        case WebCore::CryptoAlgorithmIdentifier::SHA_384:
+        case WebCore::CryptoAlgorithmIdentifier::SHA_512: {
+
+            hash = *identifier;
+            break;
+        }
+        default: {
+            JSC::throwTypeError(globalObject, scope, "digest not allowed"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        }
+    }
+
+    switch (id) {
+    case CryptoKeyClass::HMAC: {
+        const auto& hmac = downcast<WebCore::CryptoKeyHMAC>(wrapped);
+        auto result = (customHash) ? WebCore::CryptoAlgorithmHMAC::platformSignWithAlgorithm(hmac, hash, vectorData) : WebCore::CryptoAlgorithmHMAC::platformSign(hmac, vectorData);
+        if (result.hasException()) {
+            WebCore::propagateException(*globalObject, scope, result.releaseException());
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        auto resultData = result.releaseReturnValue();
+        auto size = resultData.size();
+        auto* buffer = jsCast<JSUint8Array*>(JSValue::decode(JSBuffer__bufferFromLength(globalObject, size)));
```

Comment:
This will memset(ptr, 0, len) and it doesn't need to. There is a way to pass JSC directly the data and it will do the memcpy for us without the extra unnecessary zero'ing

but this is okay for now